core/sql/executor/ExHbaseSelect.cpp

```diff
@@ -444,6 +444,7 @@ ExWorkProcRetcode ExHbaseScanSQTaskTcb::work(short &rc)
   Lng32 retcode = 0;
   rc = 0;
   Lng32 remainingInBatch = batchSize_;
+  NABoolean isFirstBatch = false;
```

Comment:
I think you are right , I thought the only exit would be due to batch size exceeding, and was seeing batch size =*K, so was good. But now I see  	    if (tcb_->moveRowToUpQueue(tcb_->convertRow_, tcb_->convertRowLen_, 
 				       &rc, FALSE))		 				       &rc, FALSE))
 	      return 1;		 	      return 1;

I am guessing this is what you are talking about (parent queue fill up). Can this happen realistically before the first 8K?